File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
     unordered_map<string,vector<string>>mp;
     for(string a:strs){
        string t=a;
        sort(t.begin(),t.end());
        mp[t].push_back(a);
     }   
     vector<vector<string>>ang;
     for(auto p:mp){
        ang.push_back(p.second);
     }
     return ang;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        ["eat","tea","tan","ate","nat","bat"]
        hash_map={}
        for s in strs:
            temp="".join(sorted(s))
            if temp in hash_map:
                hash_map[temp].append(s)
            else:
                hash_map[temp]=[s]
        return list(hash_map.values())
        